packages/common/src/common/toast.tsx

```diff
@@ -0,0 +1,217 @@
+    border-radius: 0;
+    // HDS doesn't render the icon if the toast doesn't have a label, so we
+    // show the toastify icon instead, and make room for it with padding
+    &:not(:has([class*="Notification-module_label"])) {
+      [class^="Toastify__toast-icon"] {
+        display: block;
+      }
+      [class*="Notification-module_notification"] {
+        padding-left: var(--spacing-2-xl);
+        padding-right: var(--spacing-2-xl);
+      }
+    }
+  }
+  [class^="Toastify__close-button"] {
+    position: absolute;
+    top: var(--spacing-s);
+    right: var(--spacing-s);
+    width: var(--spacing-m);
+    height: var(--spacing-m);
+    background: black;
+    mask-image: url("data:image/svg+xml;charset=utf-8,%3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 24 24'%3E%3Cpath fill='currentColor' fill-rule='evenodd' d='M18 7.5 13.5 12l4.5 4.5-1.5 1.5-4.5-4.5L7.5 18 6 16.5l4.5-4.5L6 7.5 7.5 6l4.5 4.5L16.5 6 18 7.5z' clip-rule='evenodd'/%3E%3C/svg%3E");
+  }
+  [class^="Toastify__progress-bar--wrp"] {
+    top: 0;
+    bottom: auto;
+    height: 8px;
+  }
+  [class^="Toastify__progress-bar--bg"] {
+    background: transparent;
+  }
+  [class^="Toastify__progress-bar "] {
+    opacity: 1;
+  }
+`;
+
+const HDSNotification = styled(Notification)`
+  && {
+    // We are using the border as progress bar, so we need to hide the default border
+    border-width: 0;
+  }
+  [class^="Notification-module_content"] {
+    padding-top: 8px;
+  }
+  .progress-bar {
+    position: absolute;
+    top: 0;
+    left: 0;
+    width: 100%;
+    height: 8px;
+    background-color: var(--notification-border-color);
+  }
+`;
+
+// any component can use this function to show a toast, as long as <ToastContainer /> is rendered somewhere in the app
+export default function toast({
+  label,
+  text,
+  type = "info",
+  duration,
+  dataTestId,
+  ariaLabel = label,
+  options,
+}: ToastProps): Id {
+  const toastOptions = options || {};
+  if (duration) {
+    toastOptions.autoClose = duration * 1000;
+  }
+  // toastify uses "warning" instead of "alert", so we need to convert it from HDS to toastify
+  toastOptions.type = type === "alert" ? "warning" : type;
+  switch (type) {
+    case "error":
+      toastOptions.icon = <IconErrorFill />;
+      break;
+    case "success":
+      toastOptions.icon = <IconCheckCircleFill />;
+      break;
+    default:
+      toastOptions.icon = <IconInfoCircleFill />;
+  }
+  return toastFn(
+    <ToastNotification
+      text={text}
+      type={type ?? "info"}
+      label={label}
+      ariaLabel={ariaLabel}
+      duration={duration && duration * 1000}
+      dataTestId={dataTestId}
+    />,
+    toastOptions
+  );
+}
+
+// This component is used to render the toast content, in our case it's a HDS Notification
+function ToastNotification({
+  type,
+  text,
+  label,
+  duration,
+  ariaLabel,
+  dataTestId,
+}: ToastProps): JSX.Element {
+  return (
+    <HDSNotification
+      type={type as NotificationType}
```

Comment:
this doesn't seem correct. is there a key difference between `type` and HDS notification `type`? that should be transformed. there is a conversion for it above